jbpm-test-coverage/src/test/java/org/jbpm/test/regression/async/AsyncWIHOnOracleTest.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2015 JBoss Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jbpm.test.regression.async;
+
+import org.assertj.core.api.Assertions;
+import org.jbpm.executor.impl.wih.AsyncWorkItemHandler;
+import org.jbpm.persistence.jpa.hibernate.DisabledFollowOnLockOracle10gDialect;
+import org.jbpm.persistence.util.PersistenceUtil;
+import org.jbpm.test.JbpmAsyncJobTestCase;
+import org.junit.After;
+import org.junit.Test;
+import org.kie.api.executor.Command;
+import org.kie.api.executor.CommandContext;
+import org.kie.api.executor.ExecutionResults;
+import org.kie.api.runtime.KieSession;
+import org.kie.api.runtime.process.WorkItemManager;
+import qa.tools.ikeeper.annotation.BZ;
+
+public class AsyncWIHOnOracleTest extends JbpmAsyncJobTestCase {
+
+    public static final String PROCESS = "org/jbpm/test/regression/async/AsyncWIHOnOracle.bpmn2";
+    public static final String PROCESS_ID = "org.jbpm.test.regression.async.AsyncWIHOnOracle";
+
+    private static final int EXECUTOR_THREADS = 2;
+    private static final int EXECUTOR_RETRIES = 2;
+    private static final int EXECUTOR_INTERVAL = 1;
+
+    public AsyncWIHOnOracleTest() {
+        super(EXECUTOR_THREADS, EXECUTOR_RETRIES, EXECUTOR_INTERVAL);
+    }
+
+    @Override
+    public void setUp() throws Exception {
+        String driverClassName = PersistenceUtil.getDatasourceProperties().getProperty("driverClassName");
+        if (driverClassName != null && driverClassName.contains("Oracle")) {
+            String hibernateDialect = DisabledFollowOnLockOracle10gDialect.class.getName();
+            addPersistenceProperty("hibernate.dialect", hibernateDialect);
+            logger.info("Using hibernate.dialect=" + hibernateDialect);
+        }
+        super.setUp();
+    }
+
+    @Test
+    @BZ("1234592")
+    public void testAsyncWIHExecutedMoreThanOnceOnOracle() throws InterruptedException {
+        KieSession ksession = createKSession(PROCESS);
+        WorkItemManager wim = ksession.getWorkItemManager();
+        wim.registerWorkItemHandler("async", new AsyncWorkItemHandler(getExecutorService(), CounterCommand.class.getName()));
+
+        ksession.startProcess(PROCESS_ID);
+
+        Thread.sleep(10000);
+
+        Assertions.assertThat(CounterCommand.getCounter()).as("The job has not been executed").isNotEqualTo(0);
```

Comment:
This is done on purpose. You can see there are different error messages in those two assertions. The first one should notify us that nothing has been been executed (and so there is some problem with job executor in general) and the second one that more jobs has been executed (and so there is a regression).